ios/app/JitsiMeetBroadcast Extension/SampleHandler.m

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright @ 2018-present 8x8, Inc.
+ * Copyright @ 2017-2018 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "SampleHandler.h"
+#import "SocketConnection.h"
+#import "SampleUploader.h"
+#import "DarwinNotificationCenter.h"
+
+@interface SampleHandler ()
+
+@property (nonatomic, retain) SocketConnection *clientConnection;
+@property (nonatomic, retain) SampleUploader *uploader;
+
+@end
+
+@implementation SampleHandler
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self.clientConnection = [[SocketConnection alloc] initWithFilePath:self.socketFilePath];
+    [self setupConnection];
+    
+    self.uploader = [[SampleUploader alloc] initWithConnection:self.clientConnection];    
+  }
+  
+  return self;
+}
+
+- (void)broadcastStartedWithSetupInfo:(NSDictionary<NSString *,NSObject *> *)setupInfo {
+  // User has requested to start the broadcast. Setup info from the UI extension can be supplied but optional.
+  NSLog(@"broadcast started");
+  
+  [[DarwinNotificationCenter sharedInstance] postNotificationWithName:kBroadcastStartedNotification];
+  [self openConnection];
+}
+
+- (void)broadcastPaused {
+    // User has requested to pause the broadcast. Samples will stop being delivered.
+}
+
+- (void)broadcastResumed {
+    // User has requested to resume the broadcast. Samples delivery will resume.
+}
+
+- (void)broadcastFinished {
+    // User has requested to finish the broadcast.
+  [[DarwinNotificationCenter sharedInstance] postNotificationWithName:kBroadcastStoppedNotification];
+  [self.clientConnection close];
+}
+
+- (void)processSampleBuffer:(CMSampleBufferRef)sampleBuffer withType:(RPSampleBufferType)sampleBufferType {
+  static NSUInteger frameCount = 0;
+  switch (sampleBufferType) {
+    case RPSampleBufferTypeVideo:
+      if (++frameCount%3 == 0 && self.uploader.isReady) {
+        [self.uploader sendSample:sampleBuffer];
+      }
+      break;
+
+    default:
+      break;
+  }
+}
+
+// MARK: Private Methods
+
+- (NSString *)socketFilePath {
+    NSURL *sharedContainer = [[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:@"group.org.jitsi.meet.appgroup"];
```

Comment:
Can we fish it from the Plist?

Reply:
no quick way for doing this. We could try looking at the project folder structure and read it that way, but it's not safe for future versions.

Reply:
Ack.